chain/chain/src/chain.rs

```diff
@@ -3323,12 +3323,8 @@ impl Chain {
             // only for a single shard. This so far has been enough.
             let state_patch = state_patch.take();
 
-            let storage_context = StorageContext {
-                storage_data_source: StorageDataSource::Db,
-                state_patch,
-                record_storage: self
-                    .should_produce_state_witness_for_this_or_next_epoch(me, block.header())?,
```

Comment:
`should_produce_state_witness_for_this_or_next_epoch` is used in one other place in `prepare_transactions`. We should remove it from there as well and delete the function.